### PR TITLE
it: update translation

### PR DESCRIPTION
**Pull request by henrythebuilder**
Context: #537 

Update the Italian translation (po/it.po) from latest source and update first three sections: 
- Summary
- Welcome
- Day 1: Morning

Enrico

### CODE REVIEW

**Comment by detro** [comment]
@henrythebuilder thank you for this, it's quite long so it's going to take a while for me to go through it, so bear with me.

Meanwhile, could you please go through the CLA process? Please see https://github.com/google/comprehensive-rust/pull/974/checks?check_run_id=15059836270

**Comment by henrythebuilder** [comment]
> @henrythebuilder thank you for this, it's quite long so it's going to take a while for me to go through it, so bear with me.
> 
> Meanwhile, could you please go through the CLA process? Please see https://github.com/google/comprehensive-rust/pull/974/checks?check_run_id=15059836270

Thanks for response.
I did the procedure for the CLA process:  'Individual agreements' done.

**Comment by henrythebuilder** [comment]
The changes seem to be many but it should be kept in mind that some have remained the same and others have only been modified in part. It should be noted that the initial refresh procedure ('Generating the PO Template' + 'Refreshing an Existing Translation') modified the initial file on which I worked.
However, if it helps, I attach the 'patch' file for the new changes compared to the original 'regenerated' ([it-update-translation.patch.txt](https://github.com/google/comprehensive-rust/files/12064426/it-update-translation.patch.txt))

**Comment by mgeisler** [comment]
> @henrythebuilder thank you for this, it's quite long so it's going to take a while for me to go through it, so bear with me.
> 
> Meanwhile, could you please go through the CLA process? Please see https://github.com/google/comprehensive-rust/pull/974/checks?check_run_id=15059836270

@detro, just in case you haven't seen it: you can review the commits one-by-one and then you get the changes are much more manageable:

image

**Comment by henrythebuilder** [comment]
Need help:
I got news by email that the PR integration failed -> [PR run failed: Test, Attempt #2 - it: update translation (9496e96)](https://github.com/google/comprehensive-rust/actions/runs/5544681099)
Errors look like this: `'msgid' and 'msgstr' entries do not both end with '\n'`
The error seems easy to fix by inserting a '\n' where needed, but it must be taken into account that in the translation the subdivision of the pragraph in English (with '\n') does not always coincide with that of the translated language.
So I ask you for support on how to handle this issue.
thanks in advance

File: po/it.po [hunk 1]
```diff
@@ -2937,39 +2904,37 @@ msgstr ""
...
-"  andare via, quindi deve essere molto stabile. Per gli esempi precedenti, "
-"Rust\n"
-"  la comunità sta ancora lavorando per trovare la soluzione migliore --- e "
-"forse lì\n"
-"  non è un'unica \"soluzione migliore\" per alcune di queste cose.\n"
+"  essere tolto, quindi deve essere molto stabile. Per gli esempi precedenti, "
+"la comunità Rust\n"
+"  sta ancora lavorando per trovare la soluzione migliore --- e "
+"forse \n"
+"  non c''è un'unica \"soluzione migliore\" per alcune di queste cose.\n"
```
Comment (detro) [inline po/it.po:2913]: Typo: `c'è`

**Review by detro — COMMENTED** [review]
This is excellent @henrythebuilder. Thank you so much for doing this, and thanks for partitioning your work into manageable commits, so to make the review easier.

I have mostly found a few typos and maybe one or two suggestions: should be easy to address or tweak.

In regards to the error you got:

```
'msgid' and 'msgstr' entries do not both end with '\n'
```

I admit I'm very much new to the `.po` format, and I leave it to @mgeisler to suggest if the workaround you put in place is the right one. But I do agree with you that it seems odd and might end up with odd formatting (i.e. line breaks where they are not needed).

Anyway, once those small issues are solved, I'd be happy to approve.

Thanks again for this.

File: po/it.po [hunk 2]
```diff
@@ -1676,29 +1663,26 @@ msgid "### Concurrency"
...
 "go. You can then copy/paste the examples into `src/main.rs` to experiment "
 "with\n"
 "them:"
 msgstr ""
-"Avrai bisogno di una nuova configurazione della cassa e delle dipendenze "
+"Avrai bisogno di creare un nuovo progetto, con nuove dipendenze "
 "scaricate e pronte per\n"
-"andare. Puoi quindi copiare/incollare gli esempi in `src/main.rs` per "
-"sperimentare\n"
+"l'utilizzo'. Puoi quindi copiare/incollare gli esempi in `src/main.rs` per "
```
[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:1684]: Ok, fixed

File: po/it.po [hunk 3]
```diff
@@ -1917,24 +1909,24 @@ msgid ""
...
 "* `cargo`: il gestore delle dipendenze di Rust e lo strumento di "
-"compilazione. Il carico sa come\n"
-"  scarica le dipendenze ospitate su <https://crates.io> e le passerà a\n"
-"  `rustc` quando costruisci il tuo progetto. Cargo include anche un test "
-"integrato\n"
-"  runner che viene utilizzato per eseguire unit test.\n"
+"compilazione. Cargo sa come\n"
+"  scaricare le dipendenze ospitate su <https://crates.io> e le passerà a\n"
+"  `rustc` quando costruisci il tuo progetto. Cargo include anche un 'test "
+"runner'\n"
```
[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:1919]: what about re-translate the sentence:
`Cargo also comes with a built-in test runner which is used to execute unit tests.`
to:
`Cargo include anche un gestore di test integrato che viene utilizzato per eseguire i test predisposti per il progetto (_unit tests_).`

File: po/it.po [hunk 4]
```diff
@@ -2022,18 +2012,17 @@ msgstr ""
...
 "      * Dipendenze di sviluppo e gestione/memorizzazione nella cache delle "
 "dipendenze di runtime\n"
-"      * [costruisci script]\n"
+"      * [costruzione script]\n"
 "      * [installazione globale]\n"
-"      * È anche estensibile anche con plug-in di comandi secondari (come "
+"      * È anche estensibile con plug-in di comandi secondari (come "
 "[cargo clippy]).\n"
-"  * Leggi di più dal [libro ufficiale dei carichi]"
+"  * Leggi di più dall' [official Cargo Book]"
```
[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:2019]: the problem is that by changing the _[id]_ `[official Cargo Book]` to another string the implied link no longer works.
maybe the 'single' translation of `official Cargo Book` would also be useful.

File: po/it.po (hunk 1 above)
Comment (henrythebuilder) [reply po/it.po:2913]: Ok, fixed

File: po/it.po [hunk 5]
```diff
@@ -3594,10 +3582,9 @@ msgstr "# Overloading di Funzione"
 
 #: src/basic-syntax/functions-interlude.md:3
 msgid "Overloading is not supported:"
-msgstr "Overloading non è supportato:"
+msgstr "L' Overloading non è supportato:"
```
[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:3678]: Ok, fixed

File: po/it.po [hunk 6]
```diff
@@ -3705,18 +3684,16 @@ msgstr ""
 #: src/exercises/bare-metal/afternoon.md:7
 #: src/exercises/concurrency/morning.md:12
 #: src/exercises/concurrency/afternoon.md:13
-#, fuzzy
 msgid ""
 "After looking at the exercises, you can look at the [solutions] provided."
 msgstr ""
-"Dopo aver esaminato gli esercizi, puoi esaminare le [soluzioni] fornite."
+"Dopo aver esaminato gli esercizi, puoi esaminare le [solutions] fornite."
```
[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:3783]: this is the same as `official Cargo Book`, by changing the string, the implied link is lost

File: po/it.po (hunk 4 above)
[the remark this answers is not in the document]
Comment (detro) [reply po/it.po:2019]: Ah, I see the problem.

Well, in this case I understand why you did it. Maybe I'd just remove the extra space between the `'` and the `[`.

[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:2019]: ok, I'll correct it as you say :+1:

File: po/it.po (hunk 3 above)
[the remark this answers is not in the document]
Comment (henrythebuilder) [reply po/it.po:1919]: :+1:

**Comment by henrythebuilder** [comment]
I updated the branch [henrythebuilder:it-translation](https://github.com/henrythebuilder/comprehensive-rust/tree/it-translation)!

**Review by detro — APPROVED** [review]

**Comment by henrythebuilder** [comment]
Thanks

**Comment by detro** [comment]
Thank You Enrico ;)

**Comment by mgeisler** [comment]
> In regards to the error you got:
> 
> ```
> 'msgid' and 'msgstr' entries do not both end with '\n'
> ```
> 
> I admit I'm very much new to the `.po` format, and I leave it to @mgeisler to suggest if the workaround you put in place is the right one. But I do agree with you that it seems odd and might end up with odd formatting (i.e. line breaks where they are not needed).

@henrif75 is now helping with all things related to the translations.

Here, the problem is that source text (`msgid`) and translation (`msgstr`) fields don't match in the way they use newlines. It's a very simply filter implemented with `msgfmt` and it doesn't actually know about the Markdown syntax used here. For us, an extra newline is mostly harmless since we split the Markdown text on `\n\n+` anyway — but when translating a GUI program, it would looks very weird if "File" becomes "File\n" in the translation.

**Pull request merged**